sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -148,9 +148,17 @@ class SessionCatalog(
     currentDb = db
   }
 
-  def getDefaultDBPath(db: String): String = {
+  /** Get the path for creating a non-default database. */
+  def createDatabasePath(db: String, path: Option[String]): String = {
```

Comment:
I did try it, but it involves a lot of code changes. 

The major issue is the type mismatch between the `path` in `case class CreateDatabase` and the `locationUri` in `CatalogDatabase`. The first one has to be `Option[String]` since users might not provide `location` in the create table command. The second one has to be `String` since we always have the `location`. 

https://github.com/gatorsmile/spark/blob/mkdir/sql/core/src/main/scala/org/apache/spark/sql/execution/command/ddl.scala#L63

If we change the type of `locationUri` in `CatalogDatabase` to `Option[String]`, the related code changes look not clean.